packages/roosterjs-content-model-api/lib/modelApi/common/queryContentModelBlocks.ts

```diff
@@ -0,0 +1,77 @@
+import type {
+    ContentModelBlockType,
+    ReadonlyContentModelBlock,
+    ReadonlyContentModelBlockBase,
+    ReadonlyContentModelBlockGroup,
+} from 'roosterjs-content-model-types';
+
+/**
+ * Query content model blocks
+ * @param group The block group to query
+ * @param type The type of block to query
+ * @param filter Optional selector to filter the blocks
+ * @param findFirstOnly True to return the first block only, false to return all blocks
+ */
+export function queryContentModelBlocks<T extends ReadonlyContentModelBlock>(
+    group: ReadonlyContentModelBlockGroup,
+    type: T extends ReadonlyContentModelBlockBase<infer U> ? U : never,
+    filter?: (element: T) => element is T,
+    findFirstOnly?: boolean
+): T[] {
+    const elements: T[] = [];
+    for (let i = 0; i < group.blocks.length; i++) {
+        if (findFirstOnly && elements.length > 0) {
+            return elements;
+        }
+        const block = group.blocks[i];
+
+        switch (block.blockType) {
+            case 'Paragraph':
+            case 'Divider':
+            case 'Entity':
+                if (isExpectedBlockType(block, type, filter)) {
```

Comment:
@JiuqingSong, I tried not verifying this in every, but not do it is causing build errors